File: pubnub-java/src/main/java/com/pubnub/api/endpoints/DeleteMessages.java
package com.pubnub.api.endpoints;

import com.pubnub.api.models.consumer.history.PNDeleteMessagesResult;

public interface DeleteMessages extends Endpoint<PNDeleteMessagesResult>{
    DeleteMessages channels(java.util.List<String> channels);

    DeleteMessages start(Long start);

    DeleteMessages end(Long end);
}

File: pubnub-kotlin/pubnub-kotlin-api/src/jvmMain/kotlin/com/pubnub/api/endpoints/DeleteMessages.kt
package com.pubnub.api.endpoints

import com.pubnub.api.Endpoint
import com.pubnub.api.models.consumer.history.PNDeleteMessagesResult

/**
 * @see [com.pubnub.api.PubNub.deleteMessages]
 */
interface DeleteMessages : com.pubnub.kmp.endpoints.DeleteMessages, Endpoint<PNDeleteMessagesResult> {
    val channels: List<String>
    val start: Long?
    val end: Long?
}
